nomad/structs/structs.go

```diff
@@ -6562,12 +6584,74 @@ func (a *AllocMetric) ExhaustQuota(dimensions []string) {
 	a.QuotaExhausted = append(a.QuotaExhausted, dimensions...)
 }
 
+// ScoreNode is used to gather top K scoring nodes in a heap
 func (a *AllocMetric) ScoreNode(node *Node, name string, score float64) {
-	if a.Scores == nil {
-		a.Scores = make(map[string]float64)
+	if a.nodeScoreMeta == nil {
```

Comment:
Doesn't this have to be `a.nodeScoreMeta == nil || a.nodeScoreMeta.NodeID != node.ID`?

Reply:
I reset it to nil once I have the final score from the normalized scorer. But I went ahead and made this change because its safer to do it this way (since it works even if the norm scorer doesn't get invoked).